app/controllers/courses_controller.rb

```diff
@@ -171,10 +171,24 @@ def sync_roster
     if params[:include_instructors] == 'true'
       roles.append(LtiDeployment::LTI_ROLES[:instructor])
     end
-    @current_job = LtiRosterSyncJob.perform_later(deployment, @current_course,
-                                                  roles,
-                                                  can_create_users: allowed_to?(:lti_manage?, with: UserPolicy),
-                                                  can_create_roles: allowed_to?(:manage?, with: RolePolicy))
+    job_args = {}
+    job_args[:deployment] = deployment.id
+    job_args[:role_types] = roles
+    job_args[:can_create_users] = allowed_to?(:lti_manage?, with: UserPolicy)
+    job_args[:can_create_roles] = allowed_to?(:manage?, with: RolePolicy)
+    name = "#{@current_course.name}_#{root_path.tr!('/', '')}"
```

Comment:
I assume this is just a unique identifier for the job? If so, I think using the deployment id (in addition to `root_path`) is safer, since I believe we allow courses to have multiple deployments.

Also adding the name of the class `LtiRosterSyncJob` would be good in case we want to do more scheduling in the future.